fineract-provider/src/main/java/org/apache/fineract/organisation/provisioning/data/ProvisioningCriteriaData.java

```diff
@@ -21,19 +21,27 @@
 import java.io.Serializable;
 import java.util.Collection;
 import java.util.List;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
 import org.apache.fineract.accounting.glaccount.data.GLAccountData;
 import org.apache.fineract.portfolio.loanproduct.data.LoanProductData;
 
+@Data
+@NoArgsConstructor
+@Accessors(chain = true)
 @SuppressWarnings("unused")
 public final class ProvisioningCriteriaData implements Comparable<ProvisioningCriteriaData>, Serializable {
 
-    private final Long criteriaId;
-    private final String criteriaName;
-    private final String createdBy;
-    private final Collection<LoanProductData> loanProducts;
+    private static final long serialVersionUID = 1L;
+
+    private Long criteriaId;
+    private String criteriaName;
+    private String createdBy;
+    private Collection<LoanProductData> loanProducts;
     private Collection<LoanProductData> selectedLoanProducts;
-    private final Collection<ProvisioningCriteriaDefinitionData> definitions;
```

Comment:
Why not using `@RequiredArgsContructor` to initialise `final` variables? @hmg65 @galovics 
Ref - https://projectlombok.org/features/constructor

Reply:
@BLasan we are removing `final` keyword as we are introducing setters and `final` only makes sense for immutable classes.